README.md

```diff
@@ -1,2 +1,91 @@
-# bender
-A Slack concourse CI resource that can trigger jobs
+# bender Resource
+
+A  Concourse resource that can trigger any job (deployments, releases, ...) using slack.
+
+## Source Configuration
+
+* `slack_token`: *Required.* The slack token to be used. For more info check [slack](https://api.slack.com/docs/token-types#bot)
+
+* `channel`: *Required.* The channel name to be used by the bot.
+
+* `bot_name`: *Optional.*, *default `bender`* The bot name will be used to identify and filter messages. All messages must be addressed to the bot `@bot_name some message`.
+
+* `grammar`: *Optional.* if not defined bender will respond to all mentions `@bot_name`. If grammar is defined bender will respond to messages matching the regex expression. Use [python regular expression](https://docs.python.org/2/library/re.html) syntax. See [examples](#Examples) for inspiration.
+
+* `template`: *Optional.* A string that will be evaluated and written to `template_filename` can be used as an input file for further jobs in the pipeline.
+
+* `template_filename`: *Optional.* *default `template_file.txt`* the file name for a generated template.
+
+## Behavior
+
+### `check`: Check for new messages that match the rules
+
+Check will report a new version `timestamp` if messages
+
+1: Are addressed to `@bot_name`.
+
+2: If `grammar` is defined it matches the regex of the message.
+
+#### `check Parameters`
+
+Check accepts no params.
+
+### `in`: Get the message
+
+A file **bender.json** will be created with the message payload.
+
+if `template` is defined it will be evaluated and written `template_filename`. For more info on syntax check [Template section](#Template) for example usage check the [examples](#Examples)
+
+#### `in Parameters`
+
+In accepts no params.
+
+### `out`: Reply to original message thread
+
+Sends a message as a thread to the original message sent by the user.
+
+#### `out Parameters`
+
+* `path`: *Required.* The path of the resource name. To fetch original message ts
+
+* `reply`: *Required.* The content of the reply. This can be also a template.
+
+### Template
+
+Under the hood the template use python [Jinj2](http://jinja.pocoo.org/docs/2.9/).
```

Comment:
Jinj2 -> Jinja2, sentence `Under the hood the template use python` looks broken to me. I'd rephrase: `Template uses Python Jinja2 templating engine`